### PR TITLE
Handle rcpputils::find_library_path() failure.

### PR DESCRIPTION
**Pull request by hidmic**
On bad environment lookups. As shown by https://github.com/ros2/rcutils/pull/292.

### CODE REVIEW

File: rosidl_typesupport_cpp/src/type_support_dispatch.hpp [hunk 1]
```diff
@@ -51,13 +51,21 @@ get_typesupport_handle_function(
...
 
       if (!map->data[i]) {
-        char library_name[1024];
-        snprintf(
-          library_name, 1023, "%s__%s",
-          map->package_name, identifier);
-        std::string library_path = rcpputils::find_library_path(library_name);
+        std::string library_name{map->package_name};
+        library_name += "__";
+        library_name += identifier;
```
Comment (hidmic) [inline rosidl_typesupport_cpp/src/type_support_dispatch.hpp:56]: @hidmic no reason to delay a potential `std::bad_alloc` to `rcpputils::find_library_path()` call.

**Comment by hidmic** [comment]
CI up to `rosidl_typesupport_c` and  `rosidl_typesupport_cpp`. Also includes `rcl`, just in case.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12410/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7377/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10120/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12323/)

**Review by Lobotuerk — APPROVED** [review]
LGTM

**Comment by brawner** [comment]
Added --packages-select to test parameters 
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12411/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7378/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10121/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12324/)

**Comment by hidmic** [comment]
Thank you @brawner. Merging!

**Pull request merged**